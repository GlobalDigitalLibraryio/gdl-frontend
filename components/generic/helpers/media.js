import { css } from 'styled-components';

const deviceSizes = {
  desktop: 1000,
  tablet: 769,
  mobile: 768
};

export default Object.entries(deviceSizes).reduce((acc, [device, size]) => {
  acc[device] = (...args) => css`
    @media screen and (${device === 'mobile'
        ? 'max'
        : 'min'}-width: ${size}px) {
      ${css(...args)};
    }
  `;
  return acc;
}, {});
