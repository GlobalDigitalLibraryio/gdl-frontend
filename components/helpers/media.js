// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import { css } from 'styled-components';
import { constants } from 'styled-system';
import type { TaggedTemplateLiteral } from 'styled-components';

/**
 * Mobile first media template
 *
 * Keeps compability with styled-system (hence the breakpoint import)
 */

const sizes = {
  tablet: constants.breakpoints[0],
};

// Iterate through the sizes and create a media template
const media = Object.keys(sizes).reduce((acc, label) => {
  acc[label] = (...args: TaggedTemplateLiteral) => css`
    @media (min-width: ${sizes[label]}em) {
      ${css(...args)};
    }
  `;

  return acc;
}, {});

export default media;
