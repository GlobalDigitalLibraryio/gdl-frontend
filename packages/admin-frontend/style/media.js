// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import { css } from 'react-emotion';
import {TABLET_BREAKPOINT} from "./misc";

/**
 * Mobile first media template
 *
 * (Also has a mobile only query)
 *
 */

// A function returning a function :)
const query = (condition: 'min' | 'max', width: number) => (
  // $FlowFixMe Flow doesn't play nice with template literals https://github.com/facebook/flow/issues/2616
  ...args
) => css`
  @media (${condition}-width: ${width}px) {
    ${css(...args)};
  }
`;

const media = {
  mobile: query('max', TABLET_BREAKPOINT - 1),
  tablet: query('min', TABLET_BREAKPOINT)
};

export default media;
