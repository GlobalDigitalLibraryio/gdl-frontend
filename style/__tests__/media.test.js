// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import { css } from 'react-emotion';
import { getStyles } from 'jest-emotion';
import media from '../media';

// TODO: Renable after jest-emotion pushes update that fixes this bug. Currently it doesn't run
test.skip('Mobile first media template', () => {
  const templateLiteral = css`
    background: blue;
    ${media.tablet`background-red;`};
  `;

  expect(getStyles(templateLiteral)).toMatchSnapshot();
});
