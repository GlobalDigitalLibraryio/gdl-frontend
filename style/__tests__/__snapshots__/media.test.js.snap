// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobile first media template 1`] = `
Array [
  "
    background: blue;
    ",
  "
  @media (",
  "min",
  "-width: ",
  "768",
  "px) {
    ",
  "background-red;",
  ";
  }
",
  ";
  ",
]
`;
